specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_RELOAD,
+        
+        /// Context menu item descriptor for "Save as" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_AS,
+
+        /// Context menu item descriptor for "Print" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PRINT,
+
+        /// Context menu item descriptor for "Create a QR code" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CREATE_QR_CODE,
+
+        /// Context menu item descriptor for "Inspect" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_INSPECT,
+
+        /// Context menu item descriptor for "Emoji menu" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_EMOJI,
+
+        /// Context menu item descriptor for "Redo" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_REDO,
+
+        /// Context menu item descriptor for "Undo" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_UNDO,
+
+        /// Context menu item descriptor for "Cut" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUT,
+
+        /// Context menu item descriptor for "Copy" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY,
+
+        /// Context menu item descriptor for "Paste" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE,
+
+        /// Context menu item descriptor for "Paste as plain text" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE_AS_PLAIN_TEXT,
+
+        /// Context menu item descriptor for "Select all" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SELECT_ALL,
+
+        /// Context menu item descriptor for "Open link in new window" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_OPEN_LINK_NEW_WINDOW,
+
+        /// Context menu item descriptor for "Save link as" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_LINK_AS,
+
+        /// Context menu item descriptor for "Copy link" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY_LINK,
+
+        /// Context menu item descriptor for "Save image as" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_IMAGE_AS,
+
+        /// Context menu item descriptor for "Copy image" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY_IMAGE,
+
+        /// Context menu item descriptor for "Copy image link" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY_IMAGE_LINK,
+
+        /// Context menu item descriptor for "Save media as" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_MEDIA_AS,
+
+        /// Context menu item descriptor for other commands not corresponding with any of the enums above.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_OTHER,
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR;
+    
+    /// Indicates the kind of context for which the context menu was created
+    /// for the `ICoreWebView2ContextMenuInfo::get_ContextKind` method
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_KIND
```

Comment:
Should this really be `COREWEBVIEW2_CONTEXT_MENU_CONTEXT_KIND`? Because other WebView2 things may have other kinds of contexts. (E.g., web worker security context, canvas rendering context, stacking context).

Maybe this is a `COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND`, to align with other renaming?